sdk/core/Azure.Core/src/Shared/GuidUtilities.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Azure.Core.Shared;
+
+/// <summary>
+/// The purpose of this class is to provide an allocation free way to parse and write Guids on .NET Standard 2.0, taking into account
+/// endianness.
+/// </summary>
+internal static class GuidUtilities
+{
+    private const int GuidSizeInBytes = 16;
+
+    public static Guid ParseGuidBytes(ReadOnlyMemory<byte> bytes)
+    {
+        if (bytes.Length != GuidSizeInBytes)
+        {
+            return default;
```

Comment:
I wonder if we should not simply throw. There is no good reason to handle this error condition

Reply:
I'm not disagreeing. Just sharing the previous comments

https://github.com/Azure/azure-sdk-for-net/pull/33881#discussion_r1096161178

and showing this change https://github.com/Azure/azure-sdk-for-net/pull/33881/commits/0ec7a80d125576f3b09c58d7da71cb20e2b029a8

Reply:
I think especially since it is now in the Core it should either throw or be changed into a Try pattern so that the consumer of the API can decide whether to throw or not

Reply:
Yeah this was updated to throw.